src/main/java/org/apache/sling/event/impl/jobs/stats/GaugeSupport.java

```diff
@@ -150,8 +154,10 @@ private void registerWithSuffix(String suffix, int count, Gauge<Long> value) {
             metricRegistry.register(metricName, value);
             gaugeMetricNames.add(metricName);
         } catch (IllegalArgumentException e) {
-            if (queueName != null) {
+            if (queueName != null && count <= 10) {
                 registerWithSuffix(suffix, count + 1, value);
+            } else {
+                logger.debug("Failed to register suffix {} for the queue {}, attempt {}", suffix, queueName, count, e);
```

Comment:
What about logging an error if it hits `10` and logging debug in any other case? eg like so:

```suggestion
            if (queueName != null && count <= 10) {
                logger.debug("Failed to register suffix {} for the queue {}, attempt {}, retrying.", suffix, queueName, count, e);
                registerWithSuffix(suffix, count + 1, value);
            } else {
                logger.error("Failed to register suffix {} for the queue {}, attempt {}, giving up.", suffix, queueName, count, e);
```